mrp_bom_nested_bom_generator/models/mrp_nested_bom.py

```diff
@@ -0,0 +1,196 @@
+from typing import Any, Dict, Set
+
+from odoo import _, api, fields, models
+
+
+class MrpNestedBomLine(models.Model):
+    _name = "mrp.nested.bom"
+    _description = "MRP Nested BOM Line"
+
+    sequence = fields.Integer(string="Sequence", default=10)
+
+    bom_id = fields.Many2one(
+        comodel_name="mrp.bom",
+        string="MRP BOM",
+        required=True,
+    )
+
+    bom_product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
+        string="Product Template",
+        related="bom_id.product_tmpl_id",
+    )
+
+    attribute_aggregated_ids = fields.Many2many(
+        "product.attribute",
+        "attribute_aggregated_ids_ref",
+        string="Parent Product Template attribute",
+        compute="_compute_attribute_aggregated_ids",
+        store=True,
+    )
+
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
+        string="Product",
+        domain="[('id', '!=', bom_product_tmpl_id)]",
+    )
+    product_qty = fields.Float(
+        string="Qty",
+        default=1,
+        required=True,
+    )
+    uom_id = fields.Many2one(
+        "uom.uom",
+        string="Unit of Measure",
+        related="product_tmpl_id.uom_id",
+        readonly=True,
+    )
+
+    attribute_ids = fields.Many2many(
+        "product.attribute",
+        "attribute_ids_ref",
+        string="Attributes",
+    )
+
+    @api.onchange("product_tmpl_id")
+    def _onchange_product_tmpl_id(self) -> None:
+        """
+        Change attribute field at changing product template field
+        :return None
+        """
+        self.attribute_ids = self.product_tmpl_id.attribute_line_ids.mapped(
+            "attribute_id"
+        )
+
+    @api.depends("bom_id.nested_bom_ids", "bom_product_tmpl_id.attribute_line_ids")
+    def _compute_attribute_aggregated_ids(self) -> None:
+        """
+        Aggregate own attributes with parent product template ones
+        :return None
+        """
+        for rec in self:
+            default_attributes = rec.product_tmpl_id.attribute_line_ids.mapped(
+                "attribute_id"
+            )
+            parent_attributes = rec.bom_product_tmpl_id.attribute_line_ids.mapped(
+                "attribute_id"
+            )
+            rec.attribute_aggregated_ids = default_attributes | parent_attributes
+
+    @api.model
+    def create_product(self, bom) -> int:
```

Comment:
Returning a product.template() record will make this function much more extending friendly

Reply:
Resolved